applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -170,6 +170,25 @@ static void le_audio_rx_data_handler(uint8_t const *const p_data, size_t data_si
 	ret = data_fifo_pointer_first_vacant_get(&ble_fifo_rx, (void *)&iso_received, K_NO_WAIT);
 	ERR_CHK_MSG(ret, "Unable to get FIFO pointer");
 
+	uint32_t bitrate, sampling_rate, octets_per_frame;
+	static bool data_length_wrn_printed;
+
+	if (CONFIG_AUDIO_DEV == HEADSET) {
+		ret = le_audio_config_get(&bitrate, &sampling_rate, &octets_per_frame);
+		if (ret) {
+			LOG_WRN("Failed to get configs");
+		}
+
+		if ((data_size != octets_per_frame) && !data_length_wrn_printed) {
+			LOG_WRN("Data length mismatch: %d (received), %d (codec)", data_size,
```

Comment:
Nitpick, but usually we write received: %d and codec: %d

Reply:
You can also add "printed once"

Reply:
Changed printing, st. it print every 500 mismatch